src/Monticello/MethodAddition.class.st

```diff
@@ -41,7 +41,9 @@ MethodAddition >> compile: aString classified: aString1 withStamp: aString2 logS
 
 { #category : #operations }
 MethodAddition >> createCompiledMethod [
-
+	"CyrilFerlicot: Why do we need to explicitly write to the source? Why do we call the compiler?
+	
+	Can't we just call #compile:classfied: here?"
```

Comment:
My understanding is that when installing a method in a class, its source is stored in the sources file and/or? the change file (epicea thing?), therefore the full original source is needed. But it's just what I deduced from various clues in the code and I might be wrong (or right, but still very far from the truth)

Reply:
yes, it needs the source. MethodAddition is odd:, it is used by MCPackageLoader>>#basicLoadDefinitions to seperate method compilation from method installation and then notification of change to the observers.

This shows that Monticello just hacked stuff on top instead of improving the meta model of the language...